src/main/kotlin/org/rust/lang/core/completion/LookupElements.kt

```diff
@@ -48,7 +47,14 @@ fun createLookupElement(element: RsCompositeElement, scopeName: String): LookupE
             .withTailText(element.valueParameterList?.text?.replace("\\s+".toRegex(), " ") ?: "()")
             .appendTailText(element.extraTailText, true)
             .withInsertHandler handler@ { context: InsertionContext, _: LookupElement ->
-                if (context.isInUseBlock) return@handler
+                if (context.getUseItem != null) {
+                    val hasSemicolon = context.getUseItem?.lastChild!!.elementType == RsElementTypes.SEMICOLON
```

Comment:
This question mark (`context.getUseItem?`) may be avoided if you save `context.getUseItem` to local variable. It's also good for performance, because you won't be traversing PSI two times.

Here are some docs about this "control flow sensitive typing" feature of Kotlin langauge:

https://kotlinlang.org/docs/reference/null-safety.html#checking-for-null-in-conditions

Reply:
I incorporated your suggestion in the last commit.

Once again RTFM holds true.